portal-impl/src/com/liferay/portal/model/impl/LayoutImpl.java

```diff
@@ -1060,40 +1060,35 @@ public boolean isPortletEmbedded(String portletId, long groupId) {
 			if ((portalPreferences != null) &&
 				layoutTypePortlet.isCustomizable()) {
 
-				portletPreferences = ListUtil.copy(portletPreferences);
-
-				portletPreferences.addAll(
-					PortletPreferencesLocalServiceUtil.getPortletPreferences(
+				portletPreferences =
+					PortletPreferencesLocalServiceUtil.fetchPortletPreferences(
 						portalPreferences.getUserId(),
-						PortletKeys.PREFS_OWNER_TYPE_USER, getPlid()));
+						PortletKeys.PREFS_OWNER_TYPE_USER, getPlid(),
+						portletId);
+
+				if (portletPreferences == null) {
+					return false;
+				}
 			}
 		}
 
-		if (portletPreferences.isEmpty()) {
+		if (portletPreferences == null) {
 			return false;
```

Comment:
The only change besides rebasing was to keep this return and remove the duplicate return in a later commit.